File: source/apps/ios-mdoc-reader/mdocReader-ios/mdocReader/Communication/TransportManager.swift
import Foundation

public protocol TransportManager: AnyObject {
    func addListener(_ listener: TransferEvent.Listener)
    func removeListener(_ listener: TransferEvent.Listener)
    func startDeviceEngagement(source: EngagementSource)
    func sendRequest(deviceRequestBytes: Data)
    func getSessionEncryption() -> SessionEncryption?
    func stopSession()
}

public struct TransportConfig {
    public init() {}
}

File: source/sdks/poc-mso-mdoc-sdk-aos/src/main/java/org/omnione/did/sdk/mdoc/proximity/reader/communication/TransportManager.java
package org.omnione.did.sdk.mdoc.proximity.reader.communication;

import org.omnione.did.sdk.mdoc.proximity.reader.core.EngagementSource;
import org.omnione.did.sdk.mdoc.proximity.reader.core.SessionEncryption;

public interface TransportManager {
    void addListener(TransferEvent.Listener listener);
    void removeListener(TransferEvent.Listener listener);
    void startDeviceEngagement(EngagementSource source);
    void sendRequest(byte[] deviceRequestBytes);
    SessionEncryption getSessionEncryption();
    void stopSession();
}
